File: src/mysqlcookbook/recipes/select/DateSent.java
// DateSent.java: fetch rows, refer to columns by name

import java.sql.*;
import com.kitebird.mcb.Cookbook;

public class DateSent
{
  public static void main (String[] args)
  {
    Connection conn = null;

    try
    {
      conn = Cookbook.connect ();

      try
      {
//#@ _FRAG_
        Statement s = conn.createStatement ();
        s.executeQuery ("SELECT srcuser,"
                        + " DATE_FORMAT(t,'%M %e, %Y') AS date_sent"
                        + " FROM mail");
        ResultSet rs = s.getResultSet ();
        while (rs.next ())  // loop through rows of result set
        {
          String name = rs.getString ("srcuser");
          String dateSent = rs.getString ("date_sent");
          System.out.println ("user: " + name + ", date sent: " + dateSent);
        }
        rs.close ();
        s.close ();
//#@ _FRAG_
      }
      catch (Exception e)
      {
        Cookbook.printErrorMessage (e);
      }
    }
    catch (Exception e)
    {
      System.err.println ("Cannot connect to server");
    }
    finally
    {
      if (conn != null)
      {
        try
        {
          conn.close ();
        }
        catch (Exception e) { }
      }
    }
  }
}

File: src/mysqlcookbook/recipes/select/date_sent.pl
#!/usr/bin/perl
# date_sent.pl: fetch rows into hash, refer to columns by name

use strict;
use warnings;
use Cookbook;

my $dbh = Cookbook::connect ();
my $sth;
my $stmt;

#@ _FRAG_
$sth = $dbh->prepare ("SELECT srcuser,
                       DATE_FORMAT(t,'%M %e, %Y') AS date_sent
                       FROM mail");
$sth->execute ();
while (my $ref = $sth->fetchrow_hashref ())
{
  printf "user: %s, date sent: %s\n", $ref->{srcuser}, $ref->{date_sent};
}
#@ _FRAG_

$dbh->disconnect ();
